app/components/RulesDecisionGraph/RulesDecisionGraph.tsx

```diff
@@ -42,22 +44,117 @@ export default function RulesDecisionGraph({
+        }
+
+        setResultsOfSimulation(data?.result);
+        const ruleRunSchema = await getRuleRunSchema(data);
+        // Filter out properties from ruleRunSchema outputs that are also present in data.result
+        const uniqueOutputs = Object.keys(ruleRunSchema?.result?.output || {}).reduce((acc: any, key: string) => {
+          if (!(key in data?.result)) {
+            acc[key] = ruleRunSchema?.result[key];
+          }
+          return acc;
+        }, {});
+
+        setOutputsOfSimulation(uniqueOutputs);
+        return { result: data };
+      } catch (e: any) {
+        message.error("Error during simulation run: " + e);
+        console.error("Error during simulation run:", e);
+        return { result: {} };
+      }
     }
     // Reset the result if there is no contextToSimulate (used to reset the trace)
     return { result: {} };
   };
 
+  const downloadJSON = (jsonData: any, filename: string) => {
+    const jsonString = JSON.stringify(jsonData, null, 2);
+    const blob = new Blob([jsonString], { type: "application/json" });
+    const url = URL.createObjectURL(blob);
+
+    const link = document.createElement("a");
+    link.href = url;
+    link.download = filename;
+    document.body.appendChild(link);
+    link.click();
+
+    // Clean up and remove the link
+    document.body.removeChild(link);
+    URL.revokeObjectURL(url);
+  };
+
+  const handleScenarioInsertion = async () => {
+    try {
+      const jsonData = await getDocument(jsonFile);
+      const scenarios: Scenario[] = await getScenariosByFilename(jsonFile);
+      const scenarioObject = {
+        tests: scenarios.map((scenario) => ({
+          name: scenario.title || "Default name",
+          input: scenario.variables.reduce((obj, each) => {
+            obj[each.name] = each.value;
+            return obj;
+          }, {}),
+          output: scenario.expectedResults.reduce((obj, each) => {
+            obj[each.name] = each.value;
+            return obj;
+          }, {}),
+        })),
+      };
+      const updatedJSON = {
+        ...jsonData,
+        ...scenarioObject,
+      };
+      return downloadJSON(updatedJSON, jsonFile);
+    } catch (error) {
+      console.error("Error fetching JSON:", error);
+      throw error;
+    }
+  };
+
+  const interceptJSONDownload = async (event: any) => {
+    if (decisionGraphRef.current && event.target?.download === "graph.json") {
+      event.preventDefault();
+      try {
+        await handleScenarioInsertion();
+      } catch (error) {
+        console.error("Error intercepting JSON download:", error);
+      }
+    }
+  };
+
+  useEffect(() => {
+    const clickHandler = (event: any) => {
+      interceptJSONDownload(event);
+    };
+
+    document.addEventListener("click", clickHandler);
+
+    return () => {
+      document.removeEventListener("click", clickHandler);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I was worried that a separate button to download very similar files could be confusing to end-users uploading to the brms rules repository. This could be disabled/removed, but I wanted the download ability to prepare for automated scenarios testing.